haystack_experimental/components/embedders/sentence_transformers_doc_image_embedder.py

```diff
@@ -0,0 +1,215 @@
+        )
+        if serialization_dict["init_parameters"].get("model_kwargs") is not None:
+            serialize_hf_model_kwargs(serialization_dict["init_parameters"]["model_kwargs"])
+        return serialization_dict
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "SentenceTransformersDocumentImageEmbedder":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary to deserialize from.
+        :returns:
+            Deserialized component.
+        """
+        init_params = data["init_parameters"]
+        if init_params.get("device") is not None:
+            init_params["device"] = ComponentDevice.from_dict(init_params["device"])
+        deserialize_secrets_inplace(init_params, keys=["token"])
+        if init_params.get("model_kwargs") is not None:
+            deserialize_hf_model_kwargs(init_params["model_kwargs"])
+        return default_from_dict(cls, data)
+
+    def warm_up(self):
+        """
+        Initializes the component.
+        """
+        if self.embedding_backend is None:
+            self.embedding_backend = _SentenceTransformersEmbeddingBackendFactory.get_embedding_backend(
+                model=self.model,
+                device=self.device.to_torch_str(),
+                auth_token=self.token,
+                trust_remote_code=self.trust_remote_code,
+                local_files_only=self.local_files_only,
+                model_kwargs=self.model_kwargs,
+                tokenizer_kwargs=self.tokenizer_kwargs,
+                config_kwargs=self.config_kwargs,
+                backend=self.backend,
+            )
+            if self.tokenizer_kwargs and self.tokenizer_kwargs.get("model_max_length"):
+                self.embedding_backend.model.max_seq_length = self.tokenizer_kwargs["model_max_length"]
+
+    @component.output_types(documents=List[Document])
+    def run(self, documents: List[Document]):
+        """
+        Embed a list of documents.
+
+        :param documents:
+            Documents to embed.
+
+        :returns:
+            A dictionary with the following keys:
+            - `documents`: Documents with embeddings.
+        """
+        if not isinstance(documents, list) or documents and not isinstance(documents[0], Document):
+            raise TypeError(
+                "SentenceTransformersDocumentEmbedder expects a list of Documents as input."
+                "In case you want to embed a list of strings, please use the SentenceTransformersTextEmbedder."
+            )
+        if self.embedding_backend is None:
+            raise RuntimeError("The embedding model has not been loaded. Please call warm_up() before running.")
+
+        images_to_embed = []
+        for doc in documents:
+            image_path = doc.meta.get(self.meta_field_for_image_path)
+            if not image_path:
+                raise ValueError(f"Image path not found in document metadata for document {doc.id}")
```

Comment:
Yeah I wondered about this type of logic. You can see I went for a slightly different approach in `DocumentToImageContent`. 

In general I think we should think where in the indexing pipeline do we expect users to split documents into different streams (e.g. Image Docs, PDF Docs, and Non-Image Docs). For example, I chose for `DocumentToImageContent` to support all three types instead of raising errors. 

What is your take? Should we use something like a FileTypeRouter or a ConditionalRouter earlier in the Indexing pipeline and enforce that at this stage we only expect Image Documents in the Image Embedders?

Reply:
Thinking more on this, I think the `FileTypeRouter` is the right component for Indexing to separate the different files. We may want to make something like a `PDFRouter` as well, but that can come later. 

At query time (so this pertains more to my PR for `DocumentToImageContent`) I think we should make a `DocumentTypeRouter` which works like the `FileTypeRouter` except it looks at the `file_path` in the meta field to determine the mimetype. I think this is the most extensible approach when we add more modalities like audio, video, etc. Then I'd have `DocumentToImageContent` raise an error if it finds an unsupported file type similar to `ImageEmbedder`.

Reply:
If you think it's better, I would be OK with this approach.